### PR TITLE
Textarea

### PR DESCRIPTION
**Pull request by AdityaVandan**
Text area (made from scratch)

### CODE REVIEW

File: src/components/atoms/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import cn from 'classnames';
+
+import './textArea.css';
+
+const TextArea = (props: Props) => {
+  const {
+    // bools for utility
```
Comment (ankitchouhan1020) [inline src/components/atoms/TextArea/TextArea.tsx:8]: migrate these comments in type Props.

File: src/components/atoms/TextArea/TextArea.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+
+type RequiredProps = {
+}
+
+
+type DefaultProps = {
+  error: string;
+  label?: string;
```
Comment (ankitchouhan1020) [inline src/components/atoms/TextArea/TextArea.tsx:115]: DefaultProps are optional by default. pass `label: ''` in defaultProps.

File: src/components/atoms/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  labelStyle: React.CSSProperties;
+  errorStyle: React.CSSProperties;
+  showError: boolean;
+  textAreaStyle: React.CSSProperties;
+  parentDivClass: string;
+  disableCopyPaste: boolean,
+
+  onEnterPress: (e: React.KeyboardEvent) => void;
```
Comment (ankitchouhan1020) [inline src/components/atoms/TextArea/TextArea.tsx:123]: mark `e?: ..`

File: src/components/atoms/TextArea/TextArea.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+const defaultProps: DefaultProps = {
+  labelStyle: {},
+  errorStyle: {},
+  showError: false,
+  textAreaStyle: {},
+  parentDivClass: "",
+  disableCopyPaste: false,
+  error: "there's an error",
+
+  onEnterPress: () => { },
```
Comment (ankitchouhan1020) [inline src/components/atoms/TextArea/TextArea.tsx:104]: onEnterPress and onBackspace press are user-specific requirements. Not needed in my opinion. Please check standard lib implementation too.

**Review by ankitchouhan1020 — CHANGES_REQUESTED** [review]

File: src/components/atoms/TextArea/TextArea.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  onChange: () => { },
+  onEnterPress: () => { },
+  onBackspace: () => { },
+  onFocus: () => { },
+  onKeyDown: () => { },
+  onKeyPress: () => { }
+}
+
+
+type RequiredProps = {
```
[the remark this answers is not in the document]
Comment (AdityaVandan) [reply src/components/atoms/TextArea/TextArea.tsx:75]: not empty

File: src/components/atoms/TextArea/TextArea.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      <textarea
+        {...restProps}
+        className={textAreaClassName}
+        style={textAreaStyle}
+        onCopy={disableCopyPaste ? (e) => onCopy(e) : () => { }}
+        onPaste={disableCopyPaste ? (e) => onPaste(e) : () => { }}
+      />
+
+      {
+        showError &&
```
Comment (dheerajsingh123) [inline src/components/atoms/TextArea/TextArea.tsx:44]: showError and error can be only one prop. Only error prop can determine if there is an error. If error === "", then don't show error.

**Review by ankitchouhan1020 — APPROVED** [review]

**Pull request merged**